src/MetaModels/Attribute/File/FileOrder.php

```diff
@@ -0,0 +1,359 @@
+     */
+    public function getColName()
+    {
+        return $this->colName;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getMetaModel()
+    {
+        return $this->metaModel;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function get($strKey)
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws \LogicException Always, as this is not intended for real use.
+     */
+    public function set($strKey, $varValue)
+    {
+        throw new \LogicException(__METHOD__ . ' is a virtual helper attribute and not intended for use.');
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws \LogicException Always, as this is not intended for real use.
+     */
+    public function handleMetaChange($strMetaName, $varNewValue)
+    {
+        throw new \LogicException(__METHOD__ . ' is a virtual helper attribute and not intended for use.');
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws \LogicException Always, as this is not intended for real use.
+     */
+    public function destroyAUX()
+    {
+        throw new \LogicException(__METHOD__ . ' is a virtual helper attribute and not intended for use.');
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws \LogicException Always, as this is not intended for real use.
+     */
+    public function initializeAUX()
+    {
+        throw new \LogicException(__METHOD__ . ' is a virtual helper attribute and not intended for use.');
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws \LogicException Always, as this is not intended for real use.
+     */
+    public function getAttributeSettingNames()
+    {
+        throw new \LogicException(__METHOD__ . ' is a virtual helper attribute and not intended for use.');
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws \LogicException Always, as this is not intended for real use.
+     */
+    public function getFieldDefinition($arrOverrides = array())
+    {
+        throw new \LogicException(__METHOD__ . ' is a virtual helper attribute and not intended for use.');
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws \LogicException Always, as this is not intended for real use.
+     */
+    public function getItemDCA($arrOverrides = array())
+    {
+        throw new \LogicException(__METHOD__ . ' is a virtual helper attribute and not intended for use.');
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function valueToWidget($varValue)
+    {
+        return $varValue;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function widgetToValue($varValue, $itemId)
+    {
+        $varValue = $this->unserializeData($varValue);
+
+        return $varValue;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setDataFor($arrValues)
+    {
+        foreach ($arrValues as $id => $varData) {
+            if ($varData === null) {
+                $varData = array();
```

Comment:
You musst use the serialize function here, because we can't save an array into the database.

Reply:
@baumannsven and @discordier what do u think?

Reply:
In Contao 3 this is ok, the database class takes care of serializing the data.
In Contao 4 this is already converted.